kube-controllers/cmd/kube-controllers/main.go

```diff
@@ -450,7 +450,7 @@ type controllerControl struct {
 func (cc *controllerControl) InitControllers(ctx context.Context, cfg config.RunConfig, k8sClientset *kubernetes.Clientset, calicoClient client.Interface) {
 	// Create a shared informer factory to allow cache sharing between controllers monitoring the
 	// same resource.
-	factory := informers.NewSharedInformerFactory(k8sClientset, 0)
+	factory := informers.NewSharedInformerFactory(k8sClientset, 1*time.Minute)
```

Comment:
This is configuring the resync period right? If so, this will probably cause the queue to have more pressure, and you will most likely make the problem described in https://github.com/kubernetes/kubernetes/issues/115658 worse.

Since this will not cause a new list call to k8s, it will only put all the objects in the cache, including the leaked ones, on the queue once a minute.

Reply:
I initially thought setting a resync period would force a relist, but turns out that isn't the case; and it makes sense as well, since it shouldn't be necessary.

Reply:
> it doesn't force a relist but IIRC it generates a "fake" update for each object in the cache, that is an important performance change ... I only see it used to "force" a reconcilation, typically when your controller has a dependency on an external system that doesn't have watches

Yee. I was mostly talking about it in the context of this PR and https://github.com/kubernetes/kubernetes/issues/115658, and from how I read the logic changed in this PR, there are no parts of the modified code that would benefit from those extra "fake" resync events, right?

There might be other things in calicio that could benefit from it tho., but I don't know the project well enough to know. So there could be a chance that adding a resync period is a good idea

Reply:
> there are no parts of the modified code that would benefit from those extra "fake" resync events, right?

Yeah it doesn't do a relist. My thinking was that since we now rely heavily on indexes being consistent as opposed to re-querying apiserver which was pretty slow on large clusters, this adds extra "insurance" that everything will be there in a good state. In our mitigation testing on the clusters where upstream was causing problems we've seen much faster processing and resync seems to have no noticeable negative effects ¯\_(ツ)_/¯

Reply:
at scale you'll have performance problems for sure

Reply:
> there are no parts of the modified code that would benefit from those extra "fake" resync events, right?

I don't believe so. In the context of this suite of controllers in particular, with the changes you've made I think we could ignore the output of informer events entirely and just use its cache. The only thing that the events are used for right now is... populating another cache :shrug: 

I'd probably go the other way - don't send the fake resync events, remove `c.podCache` and the code that populates it from those events. I mentioned this a bit in my other comment.

Reply:
> I'd probably go the other way - don't send the fake resync events, remove c.podCache and the code that populates it from those events. I mentioned this a bit in my other comment.

Did exactly this. PTAL